src/lib/Storage/DoctrineGateway.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzSupportTools\Storage;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\FetchMode;
+use Doctrine\DBAL\ParameterType;
+use eZ\Publish\SPI\Persistence\Content\ContentInfo;
+use eZ\Publish\SPI\Persistence\Content\Type;
+
+class DoctrineGateway implements Gateway
+{
+    private const CONTENTOBJECT_TABLE = 'ezcontentobject';
+    private const USER_TABLE = 'ezuser';
+    private const CONTENT_TYPE_TABLE = 'ezcontentclass';
+    private const CONTENTOBJECT_VERSION_TABLE = 'ezcontentobject_version';
+
+    /** @var \Doctrine\DBAL\Connection */
+    private $connection;
+
+    /** @var \Doctrine\DBAL\Platforms\AbstractPlatform */
+    private $databasePlatform;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+        $this->databasePlatform = $connection->getDatabasePlatform();
+    }
+
+    public function getContentObjectsCount(bool $includeDrafts = false): int
```

Comment:
Could change logic to `bool $onlyPublished = true`, as there is also archived state (trash) right?

Reply:
I added a condition ruling out trashed items. As we present this statistic as "Published content objects count" I would prefer staying with `bool includeDrafts = false`. Changed in https://github.com/ezsystems/ez-support-tools/pull/76/commits/2709ba354dd095baa067987521e3c67bef7e60e4.